app/model/application.js

```diff
@@ -173,29 +158,33 @@ ApplicationSchema.methods.isStatus = function(query) {
   return this.status === query;
 }
 
-// Helper function for application problems
-// Returns a boolean always
-ApplicationSchema.methods.hasProblem = function(query) {
-  return this.problems.indexOf(query) >= 0;
-}
+// Runs update test for application
+// Returns Promise of null on success
+ApplicationSchema.methods.testUpdate = function() {
+  const application = this;
 
-// Sets an application problem based on status or toggles if no status
-// Returns nothing
-ApplicationSchema.methods.setProblem = function(query, isSolved = !this.hasProblem(query)) {
-  if (isSolved && !this.hasProblem(query)) {
-    this.problems.push(query);
-  } else if (!isSolved && this.hasProblem(query)) {
-    this.problems = _.pull(this.problems, query);
-  }
+  return update(application)
+  .then(messages => {
+    return application.update({
+      'issue.problems': messages,
+    });
+  })
+  .then(data => {
+    // TODO: WHY U NO UPDATE
+    if (data.nModified < 1) {
+      app.log.warn('WHY U NO UPDATE APPLICATION PROBLEMS!?!');
+    }
```

Comment:
It seems that `application.issue` is undefined. As such (unless I missed something obvious), it seems to be more of a mongoose issue with single nested schemas.